lib/ExpensiMark.d.ts

```diff
@@ -0,0 +1,118 @@
+declare type Replacement = string | ((...args: string[]) => string);
+declare type Rule = {
+    name: string;
+    process?: (textToProcess: string, replacement: Replacement) => string;
+    regex?: RegExp;
+    replacement: Replacement;
+    pre?: (input: string) => string;
+    post?: (input: string) => string;
+};
+export default class ExpensiMark {
```

Comment:
Why a `class` and not a `type` or `interface`, I think suits better for a "class" with no body functions.

Also would be good to add when to use `class` over `interface` over `type` into the TYPESCRIPT_STYLE file

Reply:
Since we're typing a JavaScript class, using a class in TS is the most appropriate way to model it. This way, we're explicitly defining class constructor, methods and properties.

You suggested to use type or interface. While these options are useful, I believe they have different use cases:

- `type`: type aliases, union/intersection types, or mapped/conditional types.
- `interface`: This is great for defining a contract or the shape of an object, but isn't ideal for modeling a class with constructor and methods.

Are there some benefits of using interfaces in such case? Maybe I'm not aware of some caveats @gedu

Reply:
About the class vs. type/interface, we don't have another option here I believe. `ExpensiMark` is used as a `class` in E/App for example and must be declared in this way.